Code/GraphMol/Substruct/SubstructUtils.cpp

```diff
@@ -55,11 +55,19 @@ bool bondCompat(const BOND_SPTR &b1, const BOND_SPTR &b2,
   PRECONDITION(b2, "bad bond");
   bool res;
   if (useQueryQueryMatches && b1->hasQuery() && b2->hasQuery()) {
-    res = static_cast<QueryBond *>(b1.get())
-              ->QueryMatch(static_cast<QueryBond *>(b2.get()));
+    res = static_cast<QueryBond *>(b1.get())->QueryMatch(
+        static_cast<QueryBond *>(b2.get()));
   } else {
     res = b1->Match(b2);
   }
+  if (res && b1->getBondType() == Bond::DATIVE &&
```

Comment:
What happens when only one ( or the query ) is dative?  Is this a match or not a match?